### PR TITLE
modify Other options and date format in EditableQuestionRe~

### PR DESCRIPTION
**Pull request by AK5000M**
Modification the Other option issue/Date format in Note Editor Form

### CODE REVIEW

File: src/components/BaseQuestionnaireResponseForm/EditableQuestionnaireResponseForm.tsx [hunk 1]
```diff
@@ -69,15 +69,29 @@ export function EditableQuestionnaireResponseForm(props: BaseQuestionnaireRespon
...
-            formValues['current-medications']?.[0]?.value?.Coding?.display,
+            formValues['current-medications']?.[0]?.value?.Coding?.code === 'Other'
+                ? formValues['current-medications-other']?.[0]?.value?.string
+                : formValues['current-medications']?.[0]?.value?.Coding?.display,
         );
         textToCopy += generateTextForField('Allergies', formValues['patient-allergies']?.[0]?.value?.string);
-        textToCopy += generateTextForField('Family History', formValues['family-hostory']?.[0]?.value?.string);
+        textToCopy += generateTextForField(
+            'Family History',
+            formValues['family-hostory']?.[0]?.value?.Coding?.code === 'Other'
```
Comment (RolandoMaradiaga) [inline src/components/BaseQuestionnaireResponseForm/EditableQuestionnaireResponseForm.tsx:79]: There is a pattern of handling 'Other' values that is repeated multiple times. Consider refactoring this into a reusable function to reduce code duplication and improve maintainability.

File: src/components/BaseQuestionnaireResponseForm/EditableQuestionnaireResponseForm.tsx [hunk 2]
```diff
@@ -321,15 +413,67 @@ export function EditableQuestionnaireResponseForm(props: BaseQuestionnaireRespon
...
+                                    formValues['patient-birth-date']?.[0]?.value?.date
+                                        ? (() => {
+                                              const [year, month, day] =
+                                                  formValues['patient-birth-date'][0].value.date.split('-');
+                                              return `${month}/${day}/${year}`;
+                                          })()
+                                        : ''
+                                }
+                                onChange={(content) => {
+                                    const dateParts = content.split('/');
```
Comment (RolandoMaradiaga) [inline src/components/BaseQuestionnaireResponseForm/EditableQuestionnaireResponseForm.tsx:426]: The logic here for formatting and parsing dates is quite repetitive and hard to read. Recommend extracting the date formatting (YYYY-MM-DD <-> MM/DD/YYYY) into utility functions. Even better, this entire field could be abstracted into a reusable DateInputField component to handle label, formatting, and setValue calls cleanly. This would greatly improve readability and maintainability.

**Review by RolandoMaradiaga — CHANGES_REQUESTED** [review]
Overall: there's a lot of business logic mixed into the JSX. Pulling these out into small, testable helpers/components will simplify the form, make it easier to test edge cases, and improve long-term maintainability.
